contracts/src/Teleporter/upgrades/tests/TeleporterOwnerUpgradeableTests.t.sol

```diff
@@ -0,0 +1,101 @@
+
+pragma solidity 0.8.18;
+
+import "forge-std/Test.sol";
+import "../TeleporterOwnerUpgradeable.sol";
+import "./TeleporterUpgradeableTests.t.sol";
+
+contract ExampleOwnerUpgradeableApp is TeleporterOwnerUpgradeable {
+    constructor(
+        address teleporterRegistryAddress
+    ) TeleporterOwnerUpgradeable(teleporterRegistryAddress) {}
+}
+
+contract TeleporterOwnerUpgradeableTest is TeleporterUpgradeableTest {
+    ExampleOwnerUpgradeableApp app;
+
+    function setUp() public virtual override {
+        TeleporterUpgradeableTest.setUp();
+        app = new ExampleOwnerUpgradeableApp(address(teleporterRegistry));
+    }
+
+    function testOwnerUpdateMinTeleporterVersion() public {
+        uint256 minTeleporterVersion = app.minTeleporterVersion();
+
+        // Check that call to update minimum Teleporter version reverts for non-owners
+        vm.prank(teleporterAddress);
+        vm.expectRevert("Ownable: caller is not the owner");
+        app.updateMinTeleporterVersion();
+
+        // Check that minimum Teleporter version was not updated
+        assertEq(app.minTeleporterVersion(), minTeleporterVersion);
+
+        _addProtocolVersion(teleporterRegistry, teleporterAddress);
+
+        // Check that call to update minimum Teleporter version succeeds for owners
+        vm.prank(address(this));
+        vm.expectEmit(true, true, true, true, address(app));
+        emit MinTeleporterVersionUpdated(
+            minTeleporterVersion,
+            minTeleporterVersion + 1
+        );
+        app.updateMinTeleporterVersion();
+
+        assertEq(app.minTeleporterVersion(), minTeleporterVersion + 1);
+    }
+
+    function testOwnerTransfer() public {
+        uint256 minTeleporterVersion = app.minTeleporterVersion();
+        _addProtocolVersion(teleporterRegistry, teleporterAddress);
+
+        // Check that call to transfer ownership reverts for non-owners
+        vm.prank(teleporterAddress);
+        vm.expectRevert("Ownable: caller is not the owner");
+        app.transferOwnership(address(0));
+
+        // Check that ownership was not transferred
+        assertEq(app.owner(), address(this));
+
+        // Check that call for non owners reverts
+        vm.prank(teleporterAddress);
+        vm.expectRevert("Ownable: caller is not the owner");
+        app.updateMinTeleporterVersion();
+
+        // Check that after ownership transfer call succeeds
+        app.transferOwnership(teleporterAddress);
+        vm.prank(teleporterAddress);
+        vm.expectEmit(true, true, true, true, address(app));
+        emit MinTeleporterVersionUpdated(
+            minTeleporterVersion,
+            minTeleporterVersion + 1
+        );
+        app.updateMinTeleporterVersion();
+
+        // Check that call with old owner reverts
+        vm.prank(address(this));
+        vm.expectRevert("Ownable: caller is not the owner");
+        app.updateMinTeleporterVersion();
+    }
+
+    function testRenounceOwnership() public {
```

Comment:
Should we have a test case where `renounceOwnership` is called successfully, and then assert that the previous owner is unable to update the min teleporter version afterwards?

Reply:
yup the last few lines check this, but lmk if this should be made more clear or separated.

Reply:
Ah my bad, I was just reading it wrong before. 👍 

One minor nit, can we have a `MOCK_OWNER_ADDRESS` value that we use for pranking & transferring ownership to in this test suite rather than using the `teleporterAddress`? I think it would make it slightly more clear and easier to reason about.